### ISSUE TITLE
Problem with `num_leaves_required` checking during the GMM training

### ISSUE THREAD

**Issue by pstanisl** [kaldi-asr/kaldi#2139]
Error while computing deltas during training tri1 model.

Run command:

```bash
$ steps/train_deltas.sh --cmd run.pl 1024 11000 ./data/original/train ./data/original/train/lang ./models/original/gmm/mono_ali ./models/original/gmm/tri_1024-deltas
steps/train_deltas.sh: accumulating tree stats
steps/train_deltas.sh: getting questions for tree-building, via clustering
steps/train_deltas.sh: building the tree
bash: line 1: 11661 Aborted                 ( build-tree --verbose=1 --max-leaves=1024 --cluster-thresh=-1 ./models/original/gmm/tri_1024-deltas/treeacc ./data/original/train/lang/phones/roots.int ./models/original/gmm/tri_1024-deltas/questions.qst ./data/original/train/lang/topo ./models/original/gmm/tri_1024-deltas/tree ) 2>> ./models/original/gmm/tri_1024-deltas/log/build_tree.log >> ./models/original/gmm/tri_1024-deltas/log/build_tree.log
run.pl: job failed, log is in ./models/original/gmm/tri_1024-deltas/log/build_tree.log
```

Error log:

```
# build-tree --verbose=1 --max-leaves=1024 --cluster-thresh=-1 ./models/original/gmm//tri_1024-deltas/treeacc ./data/original/train/lang/phones/roots.int ./models/original/gmm//tri_1024-deltas/questions.qst ./data/original/train/lang/topo ./models/original/gmm//tri_1024-deltas/tree 
# Started at Wed Jan 10 14:35:21 CET 2018
#
build-tree --verbose=1 --max-leaves=1024 --cluster-thresh=-1 ./models/original/gmm//tri_1024-deltas/treeacc ./data/original/train/lang/phones/roots.int ./models/original/gmm//tri_1024-deltas/questions.qst ./data/original/train/lang/topo ./models/original/gmm//tri_1024-deltas/tree 
LOG (build-tree[5.3.66-33b0]:main():build-tree.cc:104) Number of separate statistics is 46920
LOG (build-tree[5.3.66-33b0]:BuildTree():build-tree.cc:161) BuildTree: before building trees, map has 45 leaves.
LOG (build-tree[5.3.66-33b0]:SplitDecisionTree():build-tree-utils.cc:577) DoDecisionTreeSplit: split 979 times, #leaves now 1024
LOG (build-tree[5.3.66-33b0]:BuildTree():build-tree.cc:187) Setting clustering threshold to smallest split 787.453
VLOG[1] (build-tree[5.3.66-33b0]:BuildTree():build-tree.cc:196) After decision tree split, num-leaves = 1024, like-impr = 2.34322 per frame over 3.70123e+06 frames.
VLOG[1] (build-tree[5.3.66-33b0]:BuildTree():build-tree.cc:200) Including just phones that were split, improvement is 2.34322 per frame over 3.70123e+06 frames.
LOG (build-tree[5.3.66-33b0]:BuildTree():build-tree.cc:215) BuildTree: removed 233 leaves.
ASSERTION_FAILED (build-tree[5.3.66-33b0]:BuildTree():build-tree.cc:232) : 'num_leaves - num_removed - num_actually_removed == num_leaves_required' 

[ Stack-Trace: ]

kaldi::MessageLogger::HandleMessage(kaldi::LogMessageEnvelope const&, char const*)
kaldi::MessageLogger::~MessageLogger()
kaldi::KaldiAssertFailure_(char const*, char const*, int, char const*)
kaldi::BuildTree(kaldi::Questions&, std::vector<std::vector<int, std::allocator<int> >, std::allocator<std::vector<int, std::allocator<int> > > > const&, std::vector<int, std::allocator<int> > const&, std::vector<bool, std::allocator<bool> > const&, std::vector<bool, std::allocator<bool> > const&, std::vector<std::pair<std::vector<std::pair<int, int>, std::allocator<std::pair<int, int> > >, kaldi::Clusterable*>, std::allocator<std::pair<std::vector<std::pair<int, int>, std::allocator<std::pair<int, int> > >, kaldi::Clusterable*> > > const&, float, int, float, int, bool)
main
__libc_start_main
_start

# Accounting: time=4 threads=1
# Ended (code 134) at Wed Jan 10 14:35:25 CET 2018, elapsed time 4 seconds
```

In previous versions of Kaldi, there was no error during the training and the model was completely ok.

https://github.com/kaldi-asr/kaldi/blob/36df91d35f59cf68750258bb77127694c82f7ebf/src/tree/build-tree.cc#L231

**Comment by danpovey** [comment]
Sorry about this.  This issue was reported yesterday.  I hope @vimalmanohar can do the right fix soon; for now I've added a temporary patch to turn the assertion into a warning (this error is unlikely to make any difference to anything except to training speed on Volta GPUs).
Here is the patch (it's in master already).
https://github.com/kaldi-asr/kaldi/commit/360c96ca491c96d138212731abaca5cd3ce77114
Please follow up with the output it gives, which will help us diagnose the real problem.

**Comment by danpovey** [comment]
Fixed in #2145
